components/user_settings/display/manage_languages/manage_languages.jsx

```diff
@@ -21,15 +24,49 @@ export default class ManageLanguage extends React.Component {
 
     constructor(props) {
         super(props);
+        const locales = I18n.getLanguages();
+        const userLocale = props.locale;
+        const selectedOption = {value: locales[userLocale].value, label: locales[userLocale].name};
+        this.reactSelectContainer = React.createRef();
 
         this.state = {
             locale: props.locale,
+            selectedOption,
             isSaving: false,
+            openMenu: false,
         };
     }
 
-    setLanguage = (e) => {
-        this.setState({locale: e.target.value});
+    componentDidMount() {
+        if (this.reactSelectContainer.current) {
+            this.reactSelectContainer.current.addEventListener('keydown', this.handleContainerKeyDown);
+        }
+    }
+
+    componentWillUnmount() {
+        if (this.reactSelectContainer.current) {
+            this.reactSelectContainer.current.removeEventListener('keydown', this.handleContainerKeyDown);
+        }
+    }
+
+    handleContainerKeyDown = (e) => {
+        if (isKeyPressed(e, Constants.KeyCodes.ESCAPE) && this.state.openMenu) {
+            this.setState({openMenu: false});
```

Comment:
Shouldn't this be handled by react-select itself?

Reply:
Normally yes, but not in a modal, bootstrap closes the modal on `escape`, and `e.stopPropagation();` does not work `onKeyDown` of `react-select`.